extensions/ql-vscode/test/vscode-tests/cli-integration/variant-analysis/variant-analysis-manager.test.ts

```diff
@@ -202,5 +204,136 @@ describe("Variant Analysis Manager", () => {
+      });
+
+      it("should run a remote query that is part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql", "lib.qll"],
+          filesThatDoNotExist: [],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is not part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-no-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql"],
+          filesThatDoNotExist: ["lib.qll", "not-in-pack.ql"],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is nested inside a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack-nested/subfolder/in-pack.ql",
+          filesThatExist: ["subfolder/in-pack.ql", "otherfolder/lib.qll"],
+          filesThatDoNotExist: ["subfolder/not-in-pack.ql"],
+          qlxFilesThatExist: ["subfolder/in-pack.qlx"],
+        });
+      });
+    });
+
+    async function doVariantAnalysisTest({
+      queryPath,
+      filesThatExist,
+      qlxFilesThatExist,
+      filesThatDoNotExist,
+    }: {
+      queryPath: string;
+      filesThatExist: string[];
+      qlxFilesThatExist: string[];
+      filesThatDoNotExist: string[];
+    }) {
+      const fileUri = getFile(queryPath);
+
+      await variantAnalysisManager.runVariantAnalysis(
+        fileUri,
+        progress,
+        cancellationTokenSource.token,
+      );
+
+      expect(mockSubmitVariantAnalysis).toBeCalledTimes(1);
+      expect(executeCommandSpy).toBeCalledWith(
+        "codeQL.monitorVariantAnalysis",
+        expect.objectContaining({
+          query: expect.objectContaining({ filePath: fileUri.fsPath }),
+        }),
+      );
+
+      const request: VariantAnalysisSubmission =
+        mockSubmitVariantAnalysis.mock.calls[0][1];
+
+      const packFS = await readBundledPack(request.query.pack);
+      filesThatExist.forEach((file) => {
+        expect(packFS.fileExists(file)).toBe(true);
+      });
+      if (await cli.cliConstraints.supportsQlxRemote()) {
+        qlxFilesThatExist.forEach((file) => {
+          expect(packFS.fileExists(file)).toBe(true);
+        });
+      }
+      filesThatDoNotExist.forEach((file) => {
+        expect(packFS.fileExists(file)).toBe(false);
+      });
+
+      expect(
+        packFS.fileExists("qlpack.yml") || packFS.fileExists("codeql-pack.yml"),
+      ).toBe(true);
+
+      // depending on the cli version, we should have one of these files
+      expect(
+        packFS.fileExists("qlpack.lock.yml") ||
+          packFS.fileExists("codeql-pack.lock.yml"),
+      ).toBe(true);
+
+      const packFileName = packFS.fileExists("qlpack.yml")
+        ? "qlpack.yml"
+        : "codeql-pack.yml";
+      const qlpackContents: any = load(
```

Comment:
Do we need these `any` type annotations or is there a reason you've added them?

Reply:
This was from the previous tests. I can see if I can get rid of them.